### PR TITLE
URL Recognizer

### PR DESCRIPTION
**Pull request by anatbal**
## Change Description

Add a new regex based recognizer to detect urls. The old domain recognizer is contained within the new url recognizer.

## Issue reference

This PR fixes issue #XX

## Checklist

- [x] I have reviewed the [contribution guidelines](https://github.com/microsoft/presidio/blob/main/CONTRIBUTING.md)
- [x] I have signed the CLA
- [x] My code includes unit tests
- [x] All unit tests and lint checks pass locally
- [x] My PR contains documentation updates / additions if required

### CODE REVIEW

**Review by omri374 — DISMISSED** [review]
Looks great! Thanks.
Could you please add this entity to the supported_entities doc? https://github.com/microsoft/presidio/blob/main/docs/supported_entities.md

**Comment by anatbal** [comment]
> E2E tests are failing, please reach out if you need any help with those.

Thanks, I think I know what I missed there, working on it

File: presidio-analyzer/tests/test_url_recognizer.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        # fmt: off
+        # Valid URLs
+        ("https://www.microsoft.com/", 1, ((0, 26),), ((0.6, "max"),),),
+        ("http://www.microsoft.com/", 1, ((0, 25),), ((0.6, "max"),),),
+        ("www.microsoft.com", 1, ((0, 17),), ((0.6, "max"),),),
+        ("The url is www.microsoft.com", 1, ((11, 28),), ((0.6, "max"),),),
+        ("microsoft.com", 1, ((0, 13),), ((0.6, "max"),),),
+        # Invalid URLs
+        ("www.microsoft", 0, (), ((),),),
+        ("http://microsoft", 0, (), ((),),),
```
Comment (SharonHart) [inline presidio-analyzer/tests/test_url_recognizer.py:29]: http://microsoft.com

File: presidio-analyzer/presidio_analyzer/predefined_recognizers/url_recognizer.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        patterns = patterns if patterns else self.PATTERNS
+        context = context if context else self.CONTEXT
+        super().__init__(
+            supported_entity=supported_entity,
+            patterns=patterns,
+            context=context,
+            supported_language=supported_language,
+        )
+
+    def validate_result(self, pattern_text: str):  # noqa D102
```
Comment (omri374) [inline presidio-analyzer/presidio_analyzer/predefined_recognizers/url_recognizer.py:80]: Please add a docstring

File: presidio-analyzer/tests/test_url_recognizer.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    return ["URL"]
+
+
+@pytest.mark.parametrize(
+    "text, expected_len, expected_positions, expected_score_ranges",
+    [
+        # fmt: off
+        # Valid URLs
+        ("https://www.microsoft.com/", 1, ((0, 26),), ((0.6, "max"),),),
+        ("http://www.microsoft.com/", 1, ((0, 25),), ((0.6, "max"),),),
```
[the remark this answers is not in the document]
Comment (anatbal) [reply presidio-analyzer/tests/test_url_recognizer.py:23]: Added a test with it

**Comment by anatbal** [comment]
> Also add it to that endpoint in the api documentation

Which doc exactly?

**Comment by SharonHart** [comment]
> > Also add it to that endpoint in the api documentation
> 
> Which doc exactly?
> 

api-doc.yaml

File: e2e-tests/resources/demo_anonymized.txt [hunk 4]
```diff
@@ -3,7 +3,7 @@ Here are a few examples sentences we currently support:
 Hello, my name is <PERSON> and I live in <LOCATION>.
 My credit card number is <CREDIT_CARD> and my crypto wallet id is <CRYPTO>.
 
-On <DATE_TIME> I visited <DOMAIN_NAME> and sent an email to <EMAIL_ADDRESS>,  from the IP <IP_ADDRESS>.
+On <DATE_TIME> I visited <URL> and sent an email to <EMAIL_ADDRESS>,  from the IP <IP_ADDRESS>.
```
Comment (SharonHart) [inline e2e-tests/resources/demo_anonymized.txt:6]: If in the current state domain and url are chosen at random, won't it make this test flicker?

Comment (anatbal) [reply e2e-tests/resources/demo_anonymized.txt]: After the latest change in the validation function of the recognizer, it should be DOMAIN NAME in this test

**Review by SharonHart — APPROVED** [review]
Nice! 🥇

File: presidio-analyzer/setup.py [hunk 5]
```diff
@@ -36,6 +36,7 @@
         "tldextract",
         "pyyaml",
         "phonenumbers>=8.12",
+        "validators"
     ],
```
Comment (omri374) [inline presidio-analyzer/setup.py:40]: Don't forget to remove this

Comment (SharonHart) [reply presidio-analyzer/setup.py:40]: I think that it's used?

Comment (SharonHart) [reply presidio-analyzer/setup.py:40]: Oh, I see that you've decided to not use it after all? 
In that case also the from the pipfile

Comment (anatbal) [reply presidio-analyzer/setup.py]: Done

**Review by omri374 — APPROVED** [review]
Looks great, thanks! One thing: Could you please update the CHANGELOG?

**Comment by anatbal** [comment]
> Looks great, thanks! One thing: Could you please update the CHANGELOG?

Under 2.2.24?

**Review by omri374 — APPROVED** [review]

**Review by SharonHart — APPROVED** [review]

**Pull request merged**